http/client.go

```diff
@@ -290,7 +290,9 @@ func (c *Client) sendRateLimitedRequest(
 		if err != nil {
 			return nil, errors.Newf(err, "failed executing the request %s", URL)
 		}
-		if resp.StatusCode != http.StatusRequestEntityTooLarge || resp.Header.Get("Retry-After") == "" {
+		if (resp.StatusCode != http.StatusRequestEntityTooLarge &&
+			resp.StatusCode != http.StatusServiceUnavailable) ||
```

Comment:
The bug report has the error as having code 503. So we need to cater for that.
The docs seem to say that mitaka uses 403, so that enters the mix as well.
Old openstacks use 413, so add that to the list.
And the RFC says 429, so use that too.

IMO, the logic should be:

if statusCode in [403, 413, 429, 503] and header.Retry-After != "" then retey